pkg/cluster_agent/main.py

```diff
@@ -6,41 +6,59 @@
 from os import getenv
 from datetime import datetime, timezone
 from traceback import format_exc
-from kubernetes import client, config
-from cluster_scanner import ClusterScanner
+from kubernetes import config, client
+from cluster_scanner import ClusterScanner, ResourceScanResult
+from resource_sender import ResourceSender
 
 SCAN_INTERVAL_SECONDS = 60
 EPSAGON_TOKEN = getenv("EPSAGON_TOKEN")
-CLUSTER_NAME = getenv("EPSAGON_CLUSTER_NAME")
+DEFAULT_CLUSTER_NAME = "K8s Cluster"
+CLUSTER_NAME = getenv("EPSAGON_CLUSTER_NAME", DEFAULT_CLUSTER_NAME)
 COLLECTOR_URL = getenv(
     "EPSAGON_COLLECTOR_URL",
     "https://collector.epsagon.com/resources/v1"
 )
-logging.getLogger().setLevel(
-    logging.DEBUG if (
-        getenv("EPSAGON_DEBUG", "").lower() == 'true'
-    )
-    else logging.INFO
-)
+IS_DEBUG_MODE = getenv("EPSAGON_DEBUG", "").lower() == "true"
+logging.getLogger().setLevel(logging.DEBUG if IS_DEBUG_MODE else logging.INFO)
 
 
 def main():
     if not EPSAGON_TOKEN:
-        logging.error("Missing epsagon token!")
-        return
-    if not CLUSTER_NAME:
-        logging.error("Missing cluster name!")
+        logging.error(
+            "Missing Epsagon token. "
+            "Make sure to configure EPSAGON_TOKEN in cluster_agent_deployment.yaml"
+        )
         return
 
     config.load_incluster_config()
-    logging.debug("Loaded cluster config")
-    scanner = ClusterScanner(EPSAGON_TOKEN, CLUSTER_NAME, collector_url=COLLECTOR_URL)
-    logging.debug("cluster scanner initialized")
+    logging.info("Loaded cluster config")
+    if IS_DEBUG_MODE:
+        loaded_conf = client.configuration.Configuration.get_default_copy()
+        logging.debug(
+            "Loaded cluster configuration:\nHost: %s\n"
+            "Using SSL Cert? %s\nUsing API token? %s",
+            loaded_conf.host,
+            bool(loaded_conf.ssl_ca_cert),
+            bool(loaded_conf.api_key)
+        )
+
+    scanner = ClusterScanner()
+    resource_sender = ResourceSender(EPSAGON_TOKEN, COLLECTOR_URL)
+    logging.debug("cluster scanner & resource sender have been initialized")
     while True:
         try:
             update_time = datetime.utcnow().replace(tzinfo=timezone.utc)
             logging.debug("Scanning cluster...")
-            scanner.scan(update_time)
+            scan_result: ResourceScanResult = scanner.scan()
+            if not scan_result.cluster_version:
```

Comment:
I think it is actually better that if the scan failed it will throw the error and let who ever called it handle the error. Rather than always return a result and trying to figure out if it was a successful or not scan